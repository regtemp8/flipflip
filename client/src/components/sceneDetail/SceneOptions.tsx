import clsx from 'clsx'

import { Card, CardContent, Grid, type Theme } from '@mui/material'

import createStyles from '@mui/styles/createStyles'
import withStyles, { type WithStyles } from '@mui/styles/withStyles'

import { SDT } from 'flipflip-common'
import ImageVideoCard from '../configGroups/ImageVideoCard'
import SceneOptionCard from '../configGroups/SceneOptionCard'
import { useAppSelector } from '../../store/hooks'
import { selectAppTutorial } from '../../store/app/selectors'

const styles = (theme: Theme) =>
  createStyles({
    backdropTop: {
      zIndex: theme.zIndex.modal + 1
    },
    highlight: {
      borderWidth: 2,
      borderColor: theme.palette.secondary.main,
      borderStyle: 'solid'
    },
    disable: {
      pointerEvents: 'none'
    },
    overflow: {
      overflow: 'inherit'
    }
  })

export interface SceneOptionsProps extends WithStyles<typeof styles> {
  sceneID?: number
}

function SceneOptions(props: SceneOptionsProps) {
  const tutorial = useAppSelector(selectAppTutorial())
  const classes = props.classes
  const tutorial1 =
    tutorial === SDT.optionsLeft ||
    tutorial === SDT.timing ||
    tutorial === SDT.backForth ||
    tutorial === SDT.imageSizing ||
    tutorial === SDT.nextScene ||
    tutorial === SDT.overlays
  const tutorial2 =
    tutorial === SDT.optionsRight ||
    tutorial === SDT.imageOptions ||
    tutorial === SDT.videoOptions ||
    tutorial === SDT.weighting ||
    tutorial === SDT.sordering ||
    tutorial === SDT.ordering
  return (
    <Grid container spacing={2}>
      <Grid
        item
        xs={12}
        md={6}
        className={clsx(
          tutorial1 && clsx(classes.backdropTop, classes.disable)
        )}
      >
        <Card
          classes={{ root: classes.overflow }}
          className={clsx(tutorial === SDT.optionsLeft && classes.highlight)}
        >
          <CardContent>
            <SceneOptionCard sceneID={props.sceneID} />
          </CardContent>
        </Card>
      </Grid>
      <Grid
        item
        xs={12}
        md={6}
        className={clsx(
          tutorial2 && clsx(classes.backdropTop, classes.disable)
        )}
      >
        <Card
          className={clsx(tutorial === SDT.optionsRight && classes.highlight)}
        >
          <CardContent>
            <ImageVideoCard sceneID={props.sceneID} />
          </CardContent>
        </Card>
      </Grid>
    </Grid>
  )
}

;(SceneOptions as any).displayName = 'SceneOptions'
export default withStyles(styles)(SceneOptions as any)
