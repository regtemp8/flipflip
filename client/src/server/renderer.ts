import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
