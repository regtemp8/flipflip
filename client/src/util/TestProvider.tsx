import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
