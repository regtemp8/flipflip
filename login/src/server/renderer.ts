import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
