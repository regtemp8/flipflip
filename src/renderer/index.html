<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
