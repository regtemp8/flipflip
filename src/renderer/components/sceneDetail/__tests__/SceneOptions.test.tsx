import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import SceneOptions from "../SceneOptions";
import TestProvider from "../../../../../test/util/TestProvider";

jest.mock('../../configGroups/ImageVideoCard', () => 'ImageVideoCard');
jest.mock('../../configGroups/SceneOptionCard', () => 'SceneOptionCard');

describe("SceneOptions", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider>
        <SceneOptions
          allScenes={[]}
          allSceneGrids={[]}
          scene={null}
          tutorial={null}
          isConfig={false}
          onUpdateScene={(scene, fn) => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
