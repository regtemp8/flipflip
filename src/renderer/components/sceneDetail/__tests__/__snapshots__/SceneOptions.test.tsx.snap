// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SceneOptions should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root SceneOptions-overflow-4 css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <SceneOptionCard
          allSceneGrids={[]}
          allScenes={[]}
          onUpdateScene={[Function]}
          scene={null}
          tutorial={null}
        />
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <ImageVideoCard
          isConfig={false}
          onUpdateScene={[Function]}
          scene={null}
          tutorial={null}
        />
      </div>
    </div>
  </div>
</div>
`;
